src/basic/terminal-util.c

```diff
@@ -882,6 +893,52 @@ int terminal_set_size_fd(int fd, const char *ident, unsigned rows, unsigned cols
         return 0;
 }
 
+int proc_cmdline_tty_size(const char *tty, unsigned *ret_rows, unsigned *ret_cols) {
+        _cleanup_free_ char *rowskey = NULL, *rowsvalue = NULL, *colskey = NULL, *colsvalue = NULL;
+        unsigned rows = UINT_MAX, cols = UINT_MAX;
+        int r;
+
+        if (!ret_rows && !ret_cols)
+                return 0;
+
+        tty = skip_dev_prefix(tty);
+        if (!in_charset(tty, ALPHANUMERICAL))
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "%s contains non-alphanumeric characters", tty);
+
+        rowskey = strjoin("systemd.tty.rows.", tty);
+        if (!rowskey)
+                return -ENOMEM;
+
+        colskey = strjoin("systemd.tty.columns.", tty);
+        if (!colskey)
+                return -ENOMEM;
+
+        r = proc_cmdline_get_key_many(/* flags = */ 0,
+                                      rowskey, &rowsvalue,
+                                      colskey, &colsvalue);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to read TTY size of %s from kernel cmdline: %m", tty);
+
+        if (rowsvalue) {
+                r = safe_atou(rowsvalue, &rows);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse %s=%s: %m", rowskey, rowsvalue);
```

Comment:
Previously, the error was ignored, why now it is critical?

Reply:
Because we can ignore at the call site now, so this function can return early.

Reply:
But, cols may have valid value.

Reply:
Yes but setting only the colums is not going to give a good user experience, only when both are set does it work properly, so I don't think it's a problem to ignore the other setting if one is malformatted.

Reply:
OK. That makes sense for me.